### DISCUSSION TITLE
How to delete an application when using the apps of apps pattern?

### DISCUSSION THREAD

**Discussion by fungiboletus** [argoproj/argo-cd#12209] [Q&A] [answered]
I use the app of apps pattern and whenever I delete an application in ArgoCD using the frontend, it's immediately created again. If I remove the YAML file from the repository it is then removed from ArgoCD but the resources are not removed.

What is the best way to remove applications in ArgoCD using the apps of apps pattern and without removing resources manually?

**Comment by jannfis** [comment, marked as answer]
If you use the app-of-apps pattern, your applications are managed from Git and shouldn't be touch in an imperative way (e.g. by deleting through UI or CLI). Depending on your setup (auto-sync, self-heal, etc) you will see the behaviour as described by you.

In order for resources to be teared down when you delete the Application YAML from Git in an app-of-apps scenario, you first need to put the [resources finalizer](https://argo-cd.readthedocs.io/en/stable/user-guide/app_deletion/#about-the-deletion-finalizer) on the Application YAMLs in Git and sync it. Next time you delete that Application YAML from Git, the applications' resources will be teared down along with the Application itself.

HTH.

**Reply by fungiboletus** [reply]
Thanks for the explanation!

I think the documentation could be a bit more clear regarding the resource finaliser because I read through this section and it didn't seem relevant to me.

So if I understand well it's a two steps process: I put the resource finaliser metadata, I sync, I delete the files, and I sync.

**Reply by ghost** [reply]
Yes the doc makes it sounds like putting the finalizer is all you have to do, but you have to in fact sync, then delete the yaml, then sync again.

**Comment by hrishikeshdkakkad** [comment]
Can you please post an exact config example to where I should be doing this?

**Reply by ghost** [reply]
it's documented [here ](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/#cascading-deletion)

1. add the finalizer,
2. save & commit & sync
3. delete the app definition yaml
4. save & commit & sync
